automation/build-bin.ts

```diff
@@ -174,6 +203,29 @@ async function renameInstallerFiles() {
 	}
 }
 
+/**
+ * If the CSC_LINK and CSC_KEY_PASSWORD env vars are set, digitally sign the
+ * executable installer by running the balena-io/scripts/shared/sign-exe.sh
+ * script (which must be in the PATH) using a MSYS2 bash shell.
+ */
+async function signWindowsInstaller() {
```

Comment:
Why not:

```typescript
import { execFile } from 'child_process';
import { promisify } from 'util';

const execFileAsync = promisify(execFile);

await execFileAsync(
    'C:\\msys64\\usr\\bin\\bash.exe',
    ['sign-exe.sh, '-f', exeName, '-d', `balena-cli ${version}`],
)
```
?
You don't need to escape anything this way.

Reply:
Thanks. I understand that the main advantage of this suggestion is to avoid the `bash -c` argument, which runs commands from a string and creates the need for escaping arguments with the `shell-escape` package.  This suggestion may work for running shell scripts like `sign-exe.sh`, but the `runUnderMsys()` function also runs other non-shell executables, like re-running `npm run build:installer` in a MSYS2 bash "environment" (`runUnderMsys()` is also called by the `run()` function in `run.ts`). In that case, `bash -c` is required, and escaping becomes necessary.
So yes, the suggestion would probably work well for `signWindowsInstaller()`, but we would still need to use `shell-escape` in `runUnderMsys()` for other cases. For this reason, I'd just leave `signWindowsInstaller()` as it is.

Reply:
@zvin, I am convinced that you jinxed it! :-) Calling `runUnderMsys()` to sign the Windows installer had been working fine for a while (2 weeks?), until about the time when posted this comment. Then, otherwise inexplicably, it started failing: apparently the PATH env var was somehow between modified in the child process, so that the `sign-exe.sh` could no longer find `signtool.exe`. But invoking bash via `execFile`, as you suggested, prevented the PATH modification and it's now working again. So I've adopted your suggestion because it is actually fixing a problem!